### ISSUE TITLE
MD052 triggered by selected checkbox in github flavoured markdown

### ISSUE THREAD

**Issue by deosrc** [DavidAnson/markdownlint#1524]
As per the title, MD052 is being triggered by selected checkboxes in github flavoured markdown:

```markdown
- [x] Checked item
```

```text
MD052/reference-links-images Reference links and images should use a label that is defined [Missing link or image reference definition: "x"] [Context: "[x]"]
```

**Comment by deosrc** [comment]
Realised this was caused by this configuration I coped from https://github.com/DavidAnson/markdownlint/blob/main/.markdownlint.json

```json
  "reference-links-images": {
    "shortcut_syntax": true
  },
```

**Comment by DavidAnson** [comment]
Is there anything to indicate this is supposed to be a checkbox instead of being a missing reference definition as is being reported? Should every instance of `[x]` be ignored in case it could be a checkbox? For my own future reference, maybe this is only meaningful as the first thing of a list item? https://github.blog/news-insights/product-news/task-lists-in-gfm-issues-pulls-comments/

**Comment by DavidAnson** [comment]
I think this is a scenario where you should disable the rule (at least for now) - you can use inline comments to do so for just the TODO list.

**Comment by DavidAnson** [comment]
> Realised this was caused by this configuration I coped from ...

Good point! I configure it to be extra picky, so this shouldn't be seen by default. Thanks for calling that out.
